### PR TITLE
Working with Auth0

### PR DESCRIPTION
**Pull request by priyaraj7**
Hi Chris,
I worked on Auth0. Now admin can login and logout to admin portal and view the messages and delete messages.
- Next I will work on code refactor i.e moving backend admin related CRUD operation in different folder. I was also working on adding new item and updating the status of item.
- Could you please review `client/src/api/menuListApi.js` , `client/src/component/admin/AdminPage.js`, `client/src/component/admin/MessagesPage.js` and `server/src/routes/itemRoutes.js` file.
- Still I need to work on naming convention of files.

### CODE REVIEW

File: server/src/routes/itemRoutes.js [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+const jwtCheck = expressjwt({
+  secret: jwks.expressJwtSecret({
+    cache: true,
+    rateLimit: true,
+    jwksRequestsPerMinute: 5,
+    jwksUri: process.env.JWKS_URI,
+  }),
+  audience: process.env.AUDIENCE,
+  issuer: process.env.ISSUER,
+  algorithms: [process.env.ALGORITHMS],
```
Comment (chrisjaure) [inline server/src/routes/itemRoutes.js:17]: Have you documented these environment variables somewhere?

File: client/src/component/admin/AdminPage.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
 function AdminPage() {
+  const { getAccessTokenSilently } = useAuth0();
+  const [tokenData, setTokenData] = useState(null);
+  useEffect(() => {
+    (async () => {
+      const audience = process.env.REACT_APP_AUDIENCE;
+      const token = await getAccessTokenSilently({ audience });
+      setTokenData(token);
+    })();
+  });
```
Comment (chrisjaure) [inline client/src/component/admin/AdminPage.js:16]: Do you want this to happen on every render?

File: client/src/component/Header.js [hunk 3]
```diff
@@ -47,13 +49,30 @@ export const ADMIN_LINKS = [
...
+export default function Header({ links, isAdminPage }) {
   const { isOpen, onToggle } = useDisclosure();
+  const { isAuthenticated, buildLogoutUrl, buildAuthorizeUrl } = useAuth0();
+  const [authUrl, setAuthUrl] = useState("");
+  useEffect(() => {
+    (async () => {
+      setAuthUrl(
+        await (isAuthenticated
+          ? buildLogoutUrl()
+          : buildAuthorizeUrl({
```
Comment (chrisjaure) [inline client/src/component/Header.js:70]: Are these two functions async? I don't think they need to be awaited.

File: client/src/component/Header.js [hunk 4]
```diff
@@ -47,13 +49,30 @@ export const ADMIN_LINKS = [
...
     label: "Messages",
     href: "/admin/messages",
   },
 ];
-export default function Header({ links }) {
+export default function Header({ links, isAdminPage }) {
   const { isOpen, onToggle } = useDisclosure();
+  const { isAuthenticated, buildLogoutUrl, buildAuthorizeUrl } = useAuth0();
+  const [authUrl, setAuthUrl] = useState("");
+  useEffect(() => {
```
Comment (chrisjaure) [inline client/src/component/Header.js:65]: This `useEffect` and state isn't needed. You can just set a const to the auth url.

**Review by chrisjaure — CHANGES_REQUESTED** [review]
Looks good but this PR adds a lot of environment variables that should be documented somewhere so it's easier to deploy.

File: client/src/component/Header.js (hunk 3 above)
Comment (priyaraj7) [reply client/src/component/Header.js:70]: buildAuthorizeUrl returns a Promise

File: client/src/component/admin/AdminPage.js (hunk 2 above)
Comment (priyaraj7) [reply client/src/component/admin/AdminPage.js:16]: NO

**Comment by priyaraj7** [comment]
> Looks good but this PR adds a lot of environment variables that should be documented somewhere so it's easier to deploy.

I saved in google doc.

**Pull request merged**